### PR TITLE
Always write logs as JSON to file

### PR DESCRIPTION
**Pull request by binarybaron**
Related issue: https://github.com/comit-network/xmr-btc-swap/issues/849

### CODE REVIEW

**Review by da-kami — APPROVED** [review]
Couldn't that cause confusion for users looking at the logs? 
I know it helps with the UI. I don't mind merging this. Any thoughts @rishflab @thomaseizinger ?

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
What is the intent of this PR? Can you provide more context on whether or not this is a behavioural change or a refactoring?

**Comment by binarybaron** [comment]
This is a behavior change. It changes the way the logs are written to disk. Right now, the format that the logs are written to depends on whether or not the swap was started with the --json flag. To provide more consistency and allow other projects built on the cli to properly parse and access information about swaps that were started manually, it would be handy if these logs that are written to a log file were always in JSON format. That's what this pull requests intends change. Should I add a changelog entry for this?

File: swap/src/cli/tracing.rs [hunk 1]
```diff
@@ -23,18 +23,19 @@ pub fn init(debug: bool, json: bool, dir: impl AsRef<Path>, swap_id: Option<Uuid
         let file_logger = fmt::layer()
             .with_ansi(false)
             .with_target(false)
+            .json()
             .with_writer(appender);
 
         if json && debug {
```
Comment (thomaseizinger) [inline swap/src/cli/tracing.rs:29]: Testing for the `json` flag is now pointless here, isn't it?

File: swap/src/cli/tracing.rs [hunk 2]
```diff
@@ -23,18 +23,19 @@ pub fn init(debug: bool, json: bool, dir: impl AsRef<Path>, swap_id: Option<Uuid
...
             set_global_default(
                 registry
-                    .with(file_logger.json())
+                    .with(file_logger)
                     .with(debug_json_terminal_printer()),
             )?;
         } else if json && !debug {
```
Comment (thomaseizinger) [inline swap/src/cli/tracing.rs:35]: Same here.

File: swap/src/cli/tracing.rs [hunk 3]
```diff
@@ -23,18 +23,19 @@ pub fn init(debug: bool, json: bool, dir: impl AsRef<Path>, swap_id: Option<Uuid
...
             set_global_default(
                 registry
-                    .with(file_logger.json())
+                    .with(file_logger)
                     .with(info_json_terminal_printer()),
             )?;
         } else if !json && debug {
```
Comment (thomaseizinger) [inline swap/src/cli/tracing.rs:41]: And here!

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
> This is a behavioural change. It changes the way the logs are written to disk. Right now, the format that the logs are written to depends on whether or not the swap was started with the --json flag. To provide more consistency and allow other projects built on the cli to properly parse and access information about swaps that were started manually, it would be handy if these logs that are written to a log file were always in JSON format. That's what this pull requests intends change. Should I add a changelog entry for this?

Thanks, that clarifies things!

This needs a changelog entry yes.
I think there is also quite some unnecessary testing for the `json` flag now. See the inline comments. Please clean that up :)

Overall, I am okay with this change. The log file was only ever intended for post-run debugging and not something that the normal user would look at.

concept ACK but needs changelog entry and cleanup

**Comment by binarybaron** [comment]
Thanks for the feedback! I will make these changes accordingly later today.

File: swap/src/cli/tracing.rs (hunk 1 above)
Comment (binarybaron) [reply swap/src/cli/tracing.rs:29]: I don't think so. We just want the file appender to always use JSON. The default terminal printer should still adhere to whether the JSON flag was passed or not. I might be misunderstanding something here.

Comment (thomaseizinger) [reply swap/src/cli/tracing.rs:29]: We are always passing the same `file_logger` though so I think that can be extracted?

Comment (binarybaron) [reply swap/src/cli/tracing.rs:31]: Like this? Can this be simplified further?

**Review by thomaseizinger — APPROVED** [review]
Yep that makes more sense now, thanks!

bors r+

**Comment by thomaseizinger** [comment]
bors retry

**Comment by binarybaron** [comment]
bors retry

**Comment by thomaseizinger** [comment]
bors d=binarybaron

**Comment by binarybaron** [comment]
Thanks @thomaseizinger 👍

**Pull request merged**